pkg/server/nfs/nfs_server.go

```diff
@@ -133,3 +145,25 @@ func setRlimitNOFILE(logger logrus.FieldLogger) error {
 	logger.Infof("ending RLIMIT_NOFILE rlimit.Cur %d, rlimit.Max %d", rlimit.Cur, rlimit.Max)
 	return nil
 }
+
+func SetConfigParams(config string) string {
+	var (
+		tmplBuf bytes.Buffer
+		logPath string
+	)
+
+	if os.Getppid() == 1 {
+		logPath = "/proc/1/fd/1"
+	} else {
+		logPath = "/tmp/ganesha.log"
+	}
+
+	tmplVals := struct {
+		LogPath string
+	}{
+		LogPath: logPath,
+	}
+
+	template.Must(template.New("Ganesha_Config").Parse(config)).Execute(&tmplBuf, tmplVals)
+	return tmplBuf.String()
```

Comment:
Note sure if you have tested this?

What you returned here is the config string instead of the config path which will be the value of the option `-f` of `ganesha.nfsd` command, right? I think you should do in-place config file update.

Reply:
Right, since I will pass the config string as paramter to the function, so it will return replaced string.

https://github.com/longhorn/longhorn-share-manager/blob/24aea4fdaccd28b5b62aa7fcefbec7d49444d3dd/pkg/server/nfs/nfs_server.go#L119